wherehows-web/app/components/datasets/compliance/export-policy.ts

```diff
@@ -0,0 +1,172 @@
+   * @type {ComputedProperty<Array<IExportPolicyTable>>}
+   * @memberof ComplianceExportPolicy
+   */
+  datasetExportPolicy = computed('exportPolicyData', 'isEditing', function(): Array<IExportPolicyTable> {
+    const exportPolicyData: IDatasetExportPolicy = getWithDefault(this, 'exportPolicyData', {});
+
+    return policyKeys.map(key => {
+      const dataType = ExportPolicyKeys[key];
+
+      return {
+        dataType,
+        value: exportPolicyData[dataType] || false,
+        label: ExportPolicyLabels[key]
+      };
+    });
+  });
+
+  constructor() {
+    super(...arguments);
+    // Setting our default values for properties expected from parent
+    typeof this.isEditing === 'boolean' || (this.isEditing = false);
+    this.toggleEditing || (this.toggleEditing = noop);
+    this.onSaveExportPolicy || (this.onSaveExportPolicy = noop);
+  }
+
+  /**
+   * Action to mark all fields in our working copy as false for the export policy
+   */
+  @action
+  onMarkAllExportFieldsAsFalse(this: ComplianceExportPolicy): void {
+    const exportPolicy = get(this, 'datasetExportPolicy');
+
+    exportPolicy.forEach(policy => {
+      set(policy, 'value', false);
+    });
+  }
+
+  /**
+   * Action triggered by the user clicking on the button "Show More" or "Show Less" that toggles
+   * the property that controls how much of the table we are showing
+   */
+  @action
+  onShowAllExportPolicyData(this: ComplianceExportPolicy): void {
+    this.toggleProperty('shouldShowMorePolicyData');
+  }
+
+  /**
+   * Action triggered by a user edit when they click on the "Yes" or "No" radio buttons in edit
+   * mode. This allows us to record and show changes that user has made in our working copy.
+   * @param {ExportPolicyKeys} dataType - the type of DMA annotation for export policy
+   * @param {boolean} value - whether or not the dataset contains this type of data
+   */
+  @action
+  onChangeExportPolicy(dataType: ExportPolicyKeys, value: boolean): void {
+    set(get(this, 'datasetExportPolicy')[policyValues.indexOf(dataType)], 'value', value);
+  }
+
+  /**
+   * Save action that should ultimately call the POST request to update our changes to the export
+   * policy information. First, we build our POST request from the working copy
+   */
+  @action
+  saveCompliance(): void {
+    const datasetExportPolicy = get(this, 'datasetExportPolicy');
+    const exportPolicy = datasetExportPolicy.reduce(
+      (policy, datum) => {
+        policy[datum.dataType] = datum.value;
+        return policy;
+      },
+      <IDatasetExportPolicy>{}
+    );
+
+    this.onSaveExportPolicy(exportPolicy);
+  }
+
+  /**
+   * Action that helps us cancel edit mode in the parent component by calling the toggleEditing
+   * action that was passed down to us. Note, onCancel connects to the generic partial, which is why
+   * we can't just call toggleEditing directly in the template
+   */
+  @action
+  onCancel(): void {
+    this.toggleEditing(false, <any>undefined);
```

Comment:
I would type the 2nd parameter on `toggleEditing` as union with `undefined` instead. Whenever you're using `any`, think of it as `unsafe`, so only use it when no other workable option is immediately available